metal/utils.py

```diff
@@ -49,27 +55,54 @@ def __init__(
         self.best_iteration = None
         self.best_score = checkpoint_min
         self.checkpoint_runway = checkpoint_runway
+        self.checkpoint_destination = checkpoint_destination
         self.verbose = verbose
+        self.state = dict({})
+
         if checkpoint_runway and verbose:
             print(
                 f"No checkpoints will be saved in the first "
                 f"checkpoint_runway={checkpoint_runway} iterations."
             )
 
-    def checkpoint(self, model, iteration, score):
+    def checkpoint(self, model, iteration, score, optimizer, lr_scheduler):
         if iteration >= self.checkpoint_runway:
+            self.state["epoch"] = iteration
+            self.state["model"] = (model.state_dict(),)
```

Comment:
Why do these need to be tuples?

Reply:
No reason. I think it was a quick fix for a bug I ran into previously and I forgot to change it back.